src/services/github_service.py

```diff
@@ -7,46 +7,63 @@
 from github.Repository import Repository
 
 class GitHubService:
-  def __init__(self, gh_client):
+  def __init__(self, gh_client: Github):
+    """Initialize GitHub service with a client."""
     self.gh_client = gh_client
 
   def get_pr_details(self, event_path: str) -> PRDetails:
-    with open(event_path, "r") as f:
-      event_data = json.load(f)
-
-    if "issue" in event_data and "pull_request" in event_data["issue"]:
-      pull_number = event_data["issue"]["number"]
-    else:
-      pull_number = event_data["number"]
-
+    """
+    Extract pull request details from GitHub event data.
+    
+    Args:
+      event_path: Path to GitHub event JSON file
+    Returns:
+      PRDetails object containing PR information
+    """
+    event_data = self._load_event_data(event_path)
+    pull_number = self._extract_pull_number(event_data)
     repo_full_name = event_data["repository"]["full_name"]
     owner, repo = repo_full_name.split("/")
-
+    
     repo_obj = self.gh_client.get_repo(repo_full_name)
     pr = repo_obj.get_pull(pull_number)
 
     return PRDetails(owner, repo_obj.name, pull_number, pr.title, pr.body)
 
   def get_diff(self, owner: str, repo: str, pull_number: int) -> str:
-    repo_name = f"{owner}/{repo}"
-    api_url = f"https://api.github.com/repos/{repo_name}/pulls/{pull_number}"
-
+    """
+    Get the diff content for a pull request.
+    
+    Returns:
+      Diff content as string or empty string if request fails
+    """
+    api_url = f"https://api.github.com/repos/{owner}/{repo}/pulls/{pull_number}.diff"
     headers = {
       'Authorization': f'Bearer {Config.GITHUB_TOKEN}',
       'Accept': 'application/vnd.github.v3.diff'
     }
```

Comment:
Việc sử dụng `Config.GITHUB_TOKEN` để xác thực có thể tiềm ẩn rủi ro bảo mật nếu token không được bảo vệ đúng cách. Hãy đảm bảo rằng token được lưu trữ và truy xuất một cách an toàn, ví dụ như sử dụng biến môi trường hoặc các dịch vụ quản lý bí mật.